src/primitives/UnionBuilder.js

```diff
@@ -0,0 +1,37 @@
+import Immutable from 'immutable';
+
+import BaseBuilder from '../BaseBuilder';
+
+class UnionBuilder extends BaseBuilder {
+  /**
+   * Set the list of builders that make up this union. All builders should be
+   * struct types.
+   *
+   * @param {StructBuilder[]} union - List of builders in the union
+   */
+  setUnion(union) {
+    return new this.constructor(this._state.set('_unionBuilders', Immutable.List(union)));
+  }
+
+  /**
+   * Setting fields is not allowed on a UnionBuilder.
+   */
+  // eslint-disable-next-line class-methods-use-this
```

Comment:
What is this for?

Reply:
The linter expects class methods to call `this` somewhere inside the function body. I'm moving it into the linter rules and removing the check here.